src/base.config.ts

```diff
@@ -0,0 +1,198 @@
+import * as webpack from 'webpack';
+import * as path from 'path';
+import { existsSync } from 'fs';
+import CssModulePlugin from '@dojo/webpack-contrib/css-module-plugin/CssModulePlugin';
+import * as ExtractTextPlugin from 'extract-text-webpack-plugin';
+import * as ManifestPlugin from 'webpack-manifest-plugin';
+import { WebpackConfiguration } from './interfaces';
+import * as loaderUtils from 'loader-utils';
+
+const IgnorePlugin = require('webpack/lib/IgnorePlugin');
+const AutoRequireWebpackPlugin = require('auto-require-webpack-plugin');
+const slash = require('slash');
+
+const basePath = process.cwd();
+const srcPath = path.join(basePath, 'src');
+const testPath = path.join(basePath, 'tests');
+const allPaths = [srcPath, testPath];
+const mainEntry = 'main';
+const packageJsonPath = path.join(basePath, 'package.json');
+const packageJson = existsSync(packageJsonPath) ? require(packageJsonPath) : {};
+const packageName = packageJson.name || '';
+const tsLintPath = path.join(basePath, 'tslint.json');
+const tsLint = existsSync(tsLintPath) ? require(tsLintPath) : false;
+
+function getJsonpFunctionName(name: string) {
+	name = name
+		.replace(/[^a-z0-9_]/g, ' ')
+		.trim()
+		.replace(/\s+/g, '_');
+	return `dojoWebpackJsonp${name}`;
+}
+
+function getUMDCompatLoader(options: { bundles?: { [key: string]: string[] } }) {
+	const { bundles = {} } = options;
+	return {
+		loader: 'umd-compat-loader',
+		options: {
+			imports(module: string, context: string) {
+				const filePath = path.relative(basePath, path.join(context, module));
+				let chunkName = path.basename(filePath);
```

Comment:
Using `path.basename` as the chunk name will cause problems if more than one chunk has the same filename. For example, suppose an application puts its i18n bundles under `src/nls/main.ts`, with the locales object `{ fr: () => import('./fr/main') }`. In this case `main` will be used as the chunk name, which is invalid since it conflicts with the entry point. `cli-build-webpack`, I believe, uses the full path (`src/nls/fr/main`), which I'd recommend here.